contracts/math/SafeMath.sol

```diff
@@ -23,7 +25,24 @@ library SafeMath {
     }
 
     /**
-    * @dev Integer division of two numbers truncating the quotient, reverts on division by zero.
+    * @dev Multiplies two signed integers, reverts on overflow.
+    */
+    function mul(int256 a, int256 b) internal pure returns (int256) {
+        // Gas optimization: this is cheaper than requiring 'a' not being zero, but the
+        // benefit is lost if 'b' is also tested.
+        // See: https://github.com/OpenZeppelin/openzeppelin-solidity/pull/522
+        if (a == 0) {
+            return 0;
+        }
+
+        int256 c = a * b;
+        require(c / a == b && (a != -1 || b != INT256_MIN));
```

Comment:
This is somewhat hard to read, I'd split the `require` into two parts, and De Morgan the second one, like so:

```
require(!(a == -1 && b == INT256_MIN)); // explain why

int256 c = a  * b;
require(c / a == b);
```